packages/block-editor/src/components/plugin-preview/index.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalUseSlot as useSlot,
+	createSlotFill,
+	Fill,
+	MenuItem,
+} from '@wordpress/components';
+import { check } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { coreDeviceTypes } from '../preview-options';
+
+const {
+	Fill: PluginPreviewMenuFill,
+	Slot: PluginPreviewMenuSlot,
+} = createSlotFill( 'PluginPreviewMenu' );
+
+export { PluginPreviewMenuSlot };
+
+/**
+ * Component used to define a custom preview menu item and optional content.
+ *
+ * The children of this component will be displayed in the main area of the
+ * block editor, instead of the `VisualEditor` component.
+ *
+ * The `title` and `icon` are used to populate the Preview menu item.
+ *
+ * @param {Object}    props          Component properties.
+ * @param {WPElement} props.children Preview content.
+ * @param {WPIcon}    props.icon     Menu item icon to be rendered.
+ * @param {string}    props.name     A unique name of the custom preview.
+ * @param {Function}  props.onClick  Menu item click handler, e.g. for previews
+ *                                   that provide no content (`children`).
+ * @param {string}    props.title    Menu item title.
+ */
+export default function PluginPreview( {
```

Comment:
As noted in my previous comment, I don't think that this SlotFill should be located inside `@wordpress/block-editor` as this package contains APIs only related to the block editing interface. This is a higher-level concept that in fact replaces the regular interface you see for editing blocks. That's why I proposed that this functionality goes to `@wordpress/interface` that already holds similar parts of the header UI like `PinnedItems` or everything related to `ComplementaryArea` that is a internal component used by `PluginSidebar`.